Shubham_Sharma/Tic-Tac-Toe/Assets/js/Main.js

```diff
@@ -0,0 +1,167 @@
+    b9.style.backgroundColor = "#a52a2a";
+
+    b1.disabled = false;
+    b2.disabled = false;
+    b3.disabled = false;
+    b4.disabled = false;
+    b5.disabled = false;
+    b6.disabled = false;
+    b7.disabled = false;
+    b8.disabled = false;
+    b9.disabled = false;
+}
+
+//Checks for any WINNING pattern, and will call a method for highlighting those button's color
+function checker(){
+    var b1 = document.getElementById("b1").innerHTML;
+    var b2 = document.getElementById("b2").innerHTML;
+    var b3 = document.getElementById("b3").innerHTML;
+    var b4 = document.getElementById("b4").innerHTML;
+    var b5 = document.getElementById("b5").innerHTML;
+    var b6 = document.getElementById("b6").innerHTML;
+    var b7 = document.getElementById("b7").innerHTML;
+    var b8 = document.getElementById("b8").innerHTML;
+    var b9 = document.getElementById("b9").innerHTML;
+    
+
+    if(((b1=="X") || (b1=="O")) && ((b1 == b2) && (b2 == b3))){
+        highLight("b1", "b2", "b3", b1);
+    }
+    else if(((b1=="X") || (b1=="O")) && ((b1 == b4) && (b4 == b7))){
+        highLight("b1", "b4", "b7", b1);
+    }
+    else if(((b1=="X") || (b1=="O")) && ((b1 == b5) && (b5 == b9))){
+        highLight("b1", "b5", "b9", b1);
+    }
+    else if(((b2=="X") || (b2=="O")) && ((b2 == b5) && (b5 == b8))){
+        highLight("b2", "b5", "b8", b2);
+    }
+    else if(((b3=="X") || (b3=="O")) && ((b3 == b6) && (b6 == b9))){
+        highLight("b3", "b6", "b9", b3);
+    }
+    else if(((b3=="X") || (b3=="O")) && ((b3 == b5) && (b5 == b7))){
+        highLight("b3", "b5", "b7", b3);
+    }
+    else if(((b4=="X") || (b4=="O")) && ((b4 == b5) && (b5 == b6))){
+        highLight("b4", "b5", "b6", b4);
+    }
+    else if(((b7=="X") || (b7=="O")) && ((b7 == b8) && (b8 == b9))){
+        highLight("b7", "b8", "b9", b7);
+    }
+}
+
+function btnDisable(){
+    document.getElementById("b1").disabled = true;
+    document.getElementById("b2").disabled = true;
+    document.getElementById("b3").disabled = true;
+    document.getElementById("b4").disabled = true;
+    document.getElementById("b5").disabled = true;
+    document.getElementById("b6").disabled = true;
+    document.getElementById("b7").disabled = true;
+    document.getElementById("b8").disabled = true;
+    document.getElementById("b9").disabled = true;
+}
+
+//Disables the buttons, and paints the button's ORANGE, which are in WINNING pattern.
+//It takes buttons id's as the three arguments and symbol=X or O as another argument, but the symbol īs currently not in use.
+function highLight(cell1, cell2, cell3, symbol){
+    btnDisable();
+    document.getElementById(cell1).style.backgroundColor = '#FFA500';
+    document.getElementById(cell2).style.backgroundColor = '#FFA500';
+    document.getElementById(cell3).style.backgroundColor = '#FFA500';
+}
+
+//The button pressed will be disabled and symbol 'X' or 'O' will appear on it
+function setButton(button, options){
+    if(button == 1){
```

Comment:
missing DRY, code here can be reduced to 4 to 5 lines of code only

Reply:
Okay, resolved that.